pkg/hubble/peer/buffer_test.go

```diff
@@ -82,3 +82,20 @@ func TestBufferPop(t *testing.T) {
 	assert.Equal(t, 0, buf.Len())
 	assert.Equal(t, 0, buf.Cap())
 }
+
+func TestBufferPopWithClosedStopChan(t *testing.T) {
+	max := 8
+	buf := newBuffer(max)
+	assert.NotNil(t, buf)
+	assert.Equal(t, 0, buf.Len())
+	assert.Equal(t, 0, buf.Cap())
+
+	for i := 0; i < max; i++ {
+		assert.NoError(t, buf.Push(&peerpb.ChangeNotification{}))
+		assert.Equal(t, i+1, buf.Len())
+	}
+	close(buf.stop)
```

Comment:
```suggestion
	buf.Close()
```
Please, use `buf.Close()` to close the `stop` chan.

Reply:
buf.Close() cleans up b.buf, this causes b.stop chan to be performed in the first select statement in Pop();
It is not the purpose of this unit case.

Reply:
Mmh, I see with `buf.Close()` you only have a 50/50 chance to trigger the real problem whereas by artificially closing the channel, you ensure that there is a check for a closed `stop` chan.